File: code_samples/lua/security_encryption.lua
-- Security: Encryption
-- AI/ML Training Sample

Encryption = {}
Encryption.__ index = Encryption

function Encryption:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Encryption:process(input)
    self.data = input
end

function Encryption:getData()
    return self.data
end

function Encryption:validate()
    return #self.data > 0
end

-- Example usage
local instance = Encryption:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/scala/security_encryption.scala
/**
 * Security: Encryption
 * AI/ML Training Sample
 */

class Encryption {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object EncryptionExample {
  def main(args: Array[String]): Unit = {
    val instance = new Encryption()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
